x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/network/top_n_flow/query.top_n_flow_network.dsl.ts

```diff
@@ -79,11 +69,62 @@ export const buildTopNFlowQuery = ({
   return dslQuery;
 };
 
+export const buildTopNFlowCountQuery = ({
+  defaultIndex,
+  filterQuery,
+  flowTarget,
+  timerange: { from, to },
+  ip,
+}: NetworkTopNFlowCountRequestOptions): ISearchRequestParams => {
+  const filter = [...createQueryFilterClauses(filterQuery), getTimeRangeFilter(from, to)];
+  const dslQuery = {
+    allow_no_indices: true,
+    index: defaultIndex,
+    ignore_unavailable: true,
+    body: {
+      aggregations: getCountAgg(flowTarget),
+      query: {
+        bool: ip
+          ? {
+              filter,
+              should: [
+                {
+                  term: {
+                    [`${getOppositeField(flowTarget)}.ip`]: ip,
+                  },
+                },
+              ],
+              minimum_should_match: 1,
+            }
+          : {
+              filter,
+            },
+      },
+      _source: false,
+    },
+    size: 0,
+    track_total_hits: false,
+  };
+  return dslQuery;
+};
+
+const getTimeRangeFilter = (from: string, to: string) => ({
+  range: {
+    '@timestamp': { gte: from, lte: to, format: 'strict_date_optional_time' },
+  },
+});
+
+const getCountAgg = (
+  flowTarget: FlowTargetSourceDest
+): Record<string, AggregationsAggregationContainer> => ({
+  top_n_flow_count: { cardinality: { field: `${flowTarget}.ip` } },
+});
+
 const getFlowTargetAggs = (
   sort: NetworkTopNFlowRequestOptions['sort'],
   flowTarget: FlowTargetSourceDest,
   querySize: number
-) => ({
+): Record<string, AggregationsAggregationContainer> => ({
```

Comment:
When I remove the `as AggregationsAggregationContainer` assertions below I see that elastic types are erroring about `fields` value not being present in elastic aggregation types because they in fact [aren't](https://github.com/elastic/elasticsearch-js/blob/3bd7ba95f866c1ddad6f167029b7681049fec18e/src/api/types.ts#L4062). 

If we apply assertion of course it resolves the error but another problem arises, when we suppress the `top_hits` section with an assertion, one can accidentally change the values inside `top_hits` with type_check no longer being able to detect that. For example I can set `size: ''` or `_source: function() {}` or any other allowed value incorrectly later. 

We are basically trying to shove the shape of return value to match the described function return value which we put there to safeguard against type safety holes like this one :). 

So i'd suggest we better fix this.

But I am wondering why is `fields` type missing from elastic aggregation types in the first place? Are the types outdated? Or maybe `fields` shouldn't be in `top_hits` and the error is justified? Let me know.

But if in the end it's a legitimate miss on elastic types side and `fields` are legit to be used here then I guess this needs to be resolved in some way.

From the top of my mind after checking the elastic types I put together this custom interface that locally resolves those errors without sacrificing type safety so you don't need to use type assertions anymore.

```tsx
interface AggregationsAggregationContainerWithFields extends AggregationsAggregationContainer {
  aggregations?: Record<string, AggregationsAggregationContainerWithFields>;
  aggs?: Record<string, AggregationsAggregationContainerWithFields>;
  top_hits?: AggregationsTopHitsAggregation & {
    fields?: QueryDslFieldAndFormat | Field | Array<QueryDslFieldAndFormat | Field>;
  };
}
```

So you can use the type here:

```suggestion
): Record<string, AggregationsAggregationContainerWithFields> => ({
```

But of course a proper solution would be to just fix the elastic types and include `fields` shape in. 
Second best is to probably elevate this type into some shared place and add overwrtes to all 30 remaining references of `AggregationsAggregationContainer` from elastic types across codebase.

Let me know wdyt.